### PR TITLE
Fix DNS cluster's endpoint ip addr check

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by hzxuzhonghu — DISMISSED** [review]
/lgtm

File: pkg/dns/dns_test.go [hunk 1]
```diff
@@ -324,3 +323,61 @@ func (s *fakeDNSServer) setTTL(ttl uint32) {
...
+				Endpoints: []*endpointv3.LocalityLbEndpoints{
+					{
+						LbEndpoints: []*endpointv3.LbEndpoint{
+							{
+								HostIdentifier: &endpointv3.LbEndpoint_Endpoint{
+									Endpoint: &endpointv3.Endpoint{
+										Address: &v3.Address{
+											Address: &v3.Address_SocketAddress{
+												SocketAddress: &v3.SocketAddress{
+													Address: "192.168.10.244",
```
Comment (hzxuzhonghu) [inline pkg/dns/dns_test.go:341]: we need a case this is a domain

File: pkg/dns/dns_test.go [hunk 2]
```diff
@@ -363,14 +392,25 @@ func TestGetPendingResolveDomain(t *testing.T) {
...
+				clusters: []*clusterv3.Cluster{
+					&utCluster,
+				},
 			},
 			want: map[string]*pendingResolveDomain{
-				"192.168.10.244": {
-					domainName: "192.168.10.244",
-					clusters:   []*clusterv3.Cluster{utCluster[0]},
+				"192.168.2.1": {
+					domainName: "192.168.2.1",
```
Comment (hzxuzhonghu) [inline pkg/dns/dns_test.go:401]: This will need to be update, can you take over the commit from @nlgwcy if he agrees

Comment (LiZhenCheng9527) [reply pkg/dns/dns_test.go]: he agrees

File: pkg/dns/dns.go [hunk 3]
```diff
@@ -407,8 +407,8 @@ func getPendingResolveDomain(clusters []*clusterv3.Cluster) map[string]*pendingR
 					continue
 				}
 				address := socketAddr.SocketAddress.Address
-				if _, err := netip.ParseAddr(address); err != nil {
-					// This is an ip address
+				if _, err := netip.ParseAddr(address); err == nil {
+					// This is an host
```
Comment (hzxuzhonghu) [inline pkg/dns/dns.go:411]: this is an ip

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**